state/test/factory/block.go

```diff
@@ -0,0 +1,117 @@
+package factory
+
+import (
+	"time"
+
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/test/factory"
+
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+func MakeBlocks(n int, state *sm.State, privVal types.PrivValidator) []*types.Block {
+	blocks := make([]*types.Block, 0)
+
+	var (
+		prevBlock     *types.Block
+		prevBlockMeta *types.BlockMeta
+	)
+
+	appHeight := byte(0x01)
+	for i := 0; i < n; i++ {
+		height := int64(i + 1)
+
+		block, parts := makeBlockAndPartSet(*state, prevBlock, prevBlockMeta, privVal, height)
+		blocks = append(blocks, block)
+
+		prevBlock = block
+		prevBlockMeta = types.NewBlockMeta(block, parts)
+
+		// update state
+		state.AppHash = []byte{appHeight}
+		appHeight++
+		state.LastBlockHeight = height
+	}
+
+	return blocks
+}
+
+func MakeBlock(state sm.State, height int64, c *types.Commit) *types.Block {
+	block, _ := state.MakeBlock(
+		height,
+		MakeData(factory.MakeTenTxs(state.LastBlockHeight), nil, nil),
+		c,
+		state.Validators.GetProposer().Address,
+	)
+	return block
+}
+
+func MakeData(txs []types.Tx, evd []types.Evidence, msgs []types.Message) types.Data {
+	return types.Data{
+		Txs: txs,
+		Evidence: types.EvidenceData{
+			Evidence: evd,
+		},
+		Messages: types.Messages{
+			MessagesList: msgs,
+		},
+	}
+}
+
+func MakeDataFromTxs(txs []types.Tx) types.Data {
```

Comment:
[minor] Since these calls return the same value?
```go
factory.MakeDataFromTxs([]types.Tx{types.Tx("foo")})
factory.MakeData([]types.Tx{types.Tx("foo")}, nil, nil)
```
would it make sense to nuke `MakeDataFromTxs`'s implementation?

Reply:
Resolved by https://github.com/celestiaorg/celestia-core/pull/902/commits/6dff08757c1763f2db2e35020ed4f44601064d3e